kube-client/src/api/portforward.rs

```diff
@@ -138,7 +144,8 @@ impl Portforwarder {
     }
 
     /// Waits for port forwarding task to complete.
-    pub async fn join(self) -> Result<(), Error> {
+    pub async fn join(mut self) -> Result<(), Error> {
+        self.ports.clear();
```

Comment:
> Do we need to do the same for errors?

Yes I guess it makes sense to do the same for the errors.

> Which would cause compiler errors if we forget to clean up anything else that's added in the future?

Yep, also a good idea I think.